src/daoreg.cpp

```diff
@@ -389,6 +416,7 @@ void daoreg::createbuyoffer (
     storeoffer(dao_id, creator, quantity, price_per_unit, token_id, util::status_active, util::type_buy_offer);
 
   } else {
+    storeoffer(dao_id, creator, quantity, price_per_unit, token_id, util::status_active, util::type_buy_offer);
```

Comment:
I this case, the status has to be closed, because we are buying a complete offer

Reply:
done.